storage/imx6ull-flash/flashdev.c

```diff
@@ -111,13 +259,17 @@ static int flashmtd_read(struct _storage_t *strg, off_t offs, void *data, size_t
 		/* TODO: should we skip badblocks ? */
 		err = flashdrv_read(strg->dev->ctx->dma, paddr, strg->dev->ctx->databuf, meta);
 		if (err < 0) {
-			err = -EIO;
+			ret = -EIO;
 			break;
 		}
 
-		err = _flashmtd_checkECC(meta, sizeof(meta->errors) / sizeof(meta->errors[0]));
+		err = _flashmtd_checkECC(strg, paddr, sizeof(meta->errors) / sizeof(meta->errors[0]));
 		if (err < 0) {
```

Comment:
Small suggestion to remove nested condition:
```suggestion
		if (err < 0 && (err != -EUCLEAN)) {
```

Reply:
In the next line we want to save returned error (`ret = err`) wheter it's `-EUCLEAN` or not so wan't combine the checks.